frontend/src/scenes/notebooks/Nodes/NotebookNodeGroup.tsx

```diff
@@ -0,0 +1,113 @@
+import { createPostHogWidgetNode } from 'scenes/notebooks/Nodes/NodeWrapper'
+import { NotebookNodeType, PropertyFilterType, PropertyOperator } from '~/types'
+import { useActions, useValues } from 'kea'
+import { urls } from 'scenes/urls'
+import { LemonSkeleton } from 'lib/lemon-ui/LemonSkeleton'
+import { notebookNodeLogic } from './notebookNodeLogic'
+import { NotebookNodeProps } from '../Notebook/utils'
+import { useEffect } from 'react'
+import clsx from 'clsx'
+import { NotFound } from 'lib/components/NotFound'
+import { groupLogic } from 'scenes/groups/groupLogic'
+import { groupDisplayId } from 'scenes/persons/GroupActorDisplay'
+import { GroupCaption } from 'scenes/groups/Group'
+import { NodeKind } from '~/queries/schema'
+import { defaultDataTableColumns } from '~/queries/nodes/DataTable/utils'
+
+const Component = ({ attributes, updateAttributes }: NotebookNodeProps<NotebookNodeGroupAttributes>): JSX.Element => {
+    const { id, groupTypeIndex } = attributes
+
+    const logic = groupLogic({ groupKey: id, groupTypeIndex: groupTypeIndex })
+    const { groupData, groupDataLoading, groupTypeName } = useValues(logic)
+    const { setActions, insertAfter } = useActions(notebookNodeLogic)
+
+    const title = groupData
+        ? `${groupTypeName}: ${groupDisplayId(groupData.group_key, groupData.group_properties)}`
+        : 'Group'
+
+    useEffect(() => {
+        updateAttributes({
+            title,
+        })
+        setActions([
+            {
+                text: 'Events for this group',
+                onClick: () => {
+                    insertAfter({
+                        type: NotebookNodeType.Query,
+                        attrs: {
+                            title: `Events for ${title}`,
+                            query: {
+                                kind: NodeKind.DataTableNode,
+                                full: true,
+                                source: {
+                                    kind: NodeKind.EventsQuery,
+                                    select: defaultDataTableColumns(NodeKind.EventsQuery),
+                                    after: '-24h',
+                                    properties: [
+                                        {
+                                            key: `$group_${groupTypeIndex}`,
+                                            value: id,
+                                            type: PropertyFilterType.Event,
+                                            operator: PropertyOperator.Exact,
+                                        },
+                                    ],
+                                },
+                            },
+                        },
+                    })
+                },
+            },
+        ])
+    }, [groupData])
```

Comment:
I'm assuming this value changes whenever the `title` or `groupTypeIndex` does too

Reply:
This will be improved in the editable titles PR so will leave for now